### PR TITLE
Network: connection manager cleanup

### PR DESCRIPTION
**Pull request by gerardsn**
This should contain no functional changes
removes unused struct values / interfaces / tests
small refactor in start/stop to improve readability

### CODE REVIEW

File: network/transport/grpc/connection_manager.go [hunk 1]
```diff
@@ -179,13 +154,31 @@ func (s *grpcConnectionManager) Start() error {
 	serverOpts = append(serverOpts, grpc.ChainStreamInterceptor(serverInterceptors...))
 
 	// Create gRPC server for inbound connectionList and associate it with the protocols
-	s.grpcServer = grpc.NewServer(serverOpts...)
-	for _, prot := range s.protocols {
-		func(protocol Protocol) {
-			prot.Register(s, func(stream grpc.ServerStream) error {
-				return s.handleInboundStream(protocol, stream)
-			}, s.connections, s)
-		}(prot)
```
Comment (gerardsn) [inline network/transport/grpc/connection_manager.go:188]: does wrapping this in a function have any side effects?

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
With removing the `Stop()` there's no longer an option to stop the grpcServer gracefully. The connectionManager creates a background context and has to stop it on its own.

**Comment by gerardsn** [comment]
Not sure what you mean. `Stop()` is not removed, biggest change there is that it no longer sets everything to nil after closing.

**Comment by woutslakhorst** [comment]
> Not sure what you mean. `Stop()` is not removed, biggest change there is that it no longer sets everything to nil after closing.

Hmmm don't know what I was smoking...

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**